DiscountDiscover/Models/Deal.m

```diff
@@ -11,28 +11,49 @@
 @implementation Deal
 
 - (id)initWithDictionary:(NSDictionary *)dictionary {
+    
     self = [super init];
 
     NSDictionary *deal = dictionary[@"deal"];
     self.name = deal[@"title"];
-    self.description = deal[@"description"];
+    self.dealDescription = deal[@"description"];
     self.finePrint = deal[@"fine_print"];
     self.url = [NSURL URLWithString:deal[@"url"]];
     self.category = deal[@"category_name"];
     self.imageUrl = [NSURL URLWithString:deal[@"image_url"]];
-    self.expiresAt = deal[@"expiresAt"];
+    self.expiresAt = [self formatExpiresAtString:deal];
     
     NSDictionary *store = deal[@"merchant"];
     self.storeName = store[@"name"];
-    self.storeAddress = store[@"address"];
+    self.storeAddress = [NSString stringWithFormat:@"%@\n%@, %@ %@", store[@"address"], store[@"locality"], store[@"region"], store[@"postal_code"]];
```

Comment:
You will probably have to add nil checks for the different address components. Might be a good idea to add a new method to encapsulate this logic too.